### ISSUE TITLE
Add node schedule for quiz instead of checking date.now

### ISSUE THREAD

**Issue by jugaldb** [CodeChefVIT/Quizzie#107]
Currently the quiz status which is the status used to make the quiz live/closed is checked by date.now, use node schedule with cron jobs to make the update

**Comment by RishabhKodes** [comment]
> Currently the quiz status which is the status used to make the quiz live/closed is checked by date.now, use node schedule with cron jobs to make the update

@jugaldb  Add the path to the file which contains the quiz status code.

**Comment by jugaldb** [comment]
@RishabhKodes  The route is in Backend/api/routers/quiz.js  https://github.com/codechefvit/Quizzie/blob/master/Backend/api/routers/quiz.js

status = 0 , Quiz hasn't yet started
status = 1, Quiz is live
status = 2 Quiz is over

You can check the quiz timings via Backend/api/models/quiz.js
https://github.com/CodeChefVIT/Quizzie/blob/master/Backend/api/models/quiz.js

**Comment by jugaldb** [comment]
@RishabhKodes  The route name is /start in the file I mentioned above, you can very well create the whole new route and create a seperate file for cron jobs and call in that function.

**Comment by decipher07** [comment]
Hey, Can you please Assign this issue to me?

**Comment by nanipunepalle** [comment]
What exactly to be done @jugaldb .You can assign me I will give a try

**Comment by jugaldb** [comment]
> @RishabhKodes The route is in Backend/api/routers/quiz.js https://github.com/codechefvit/Quizzie/blob/master/Backend/api/routers/quiz.js
> 
> status = 0 , Quiz hasn't yet started
> status = 1, Quiz is live
> status = 2 Quiz is over
> 
> You can check the quiz timings via Backend/api/models/quiz.js
> https://github.com/CodeChefVIT/Quizzie/blob/master/Backend/api/models/quiz.js

If you can look at this, the /start route currently uses Date.now() to check the quiz timings, there is a NPM package called node-schedule, You need to schedule a job, basically a function which would run every minute and check the status of all the quizzes that exist if some quiz's scheduled for time has come, change the status from 0 -> 1 and similarly for closing, make it from 1->2

@nanipunepalle
basically in /start route change Date.now() use to node-schedule

**Comment by nanipunepalle** [comment]
@jugaldb rather than checking every minute I think we can schedule at particular time right

/start route is posted by quiz participant when he opens the website right?

**Comment by jugaldb** [comment]
> @jugaldb rather than checking every minute I think we can schedule at particular time right
> 
> /start route is posted by user when he opens the website right?

No, it is for attempting the quiz, basically remove Date.now() references and make function to automate the quiz open and close, make a new folder under name "jobs" - > "startQuiz.js" and "endQuiz.js", make the two functions in 2 different files @nanipunepalle

**Comment by jugaldb** [comment]
> @jugaldb rather than checking every minute I think we can schedule at particular time right
> 
> /start route is posted by quiz participant when he opens the website right?

There can be quiz a case where quiz is starting annytime and if in the future the number of quizzes increases, there can be one scheduled  every 5 mins, so its better do it every minute, look at cron format to do this,  @nanipunepalle

**Comment by akshatvg** [comment]
@decipher07 any updates?

**Comment by decipher07** [comment]
Working with On Schedules down in locals but not with Server, Been Searching for some more insights to this thing. will soon open a PR

**Comment by jugaldb** [comment]
> insights to this thing. will soon open a PR
Check this @decipher07 
https://stackoverflow.com/questions/20897528/schedule-a-job-for-every-5-minutes

**Comment by decipher07** [comment]
Yup, This helped with constructing the logics for the schedule. I think the PR can be merged, Just check for the route being separate from the original just to revert commits if the build fails.

**Comment by jugaldb** [comment]
This  feature has been added refer:  https://github.com/CodeChefVIT/Quizzie/pull/116 

Closing the  issue
